Add sound notification and subtitle for macOS

MatthieuMota opened the pull request:
With AppleScriptNotifier, we can provide an amazing sound and subtitle in the notification.

pyrech commented:
Hi! Thanks for your work on this PR. However, I'm not really fan of adding explicit support for options that are available only for a single notifier.

Instead of adding two properties to the Notification class I think it would be better to add a single `options` (or whatever its name). By adding an option that is not part of the Notification's public interface, it would better communicate the fact that the option you set is likely to be unsupported for most notifiers. Something like this:

```php
class Notification {
    // ...
    private $options = [];

    // ...

    public function addOption($key, $option) {
        $this->options[$key] = $option;
    }

    public function getOption($key) {
       if (!array_key_exists($key, $this->options) {
           return null;
       }
       return $this->options[$key];
    }
}

// ...
$notification->addOption('sound', 'Frog');
```

Oh and I do not agree with ignoring `idea` folder, this is something that must be configured on the developer's computer, not on every PHP projects :wink: 

Would you mind to refactor your PR this way? If you can't, no problem, I will do it later :+1:

MatthieuMota commented:
Hi, thank you for your feedback. I've refactored sound and subtitle to an array of options attribute in Notification. I've removed my commit with .idea, I agree with you for that.

pyrech commented:
Awesome! :ok_hand: I just amended your last commit to fix the CS :wink: 

Thank you!
